prime-router/src/main/kotlin/cli/tests/HistoryApiTest.kt

```diff
@@ -0,0 +1,233 @@
+                bad("***$name Test FAILED***:  response code $responseCode")
+                return null // failure
+            }
+            val reportId = getReportIdFromResponse(json)
+                ?: let {
+                    bad("***$name Test FAILED***: A report ID came back as null")
+                    return null
+                }
+            echo("Id of submitted report: $reportId")
+            reportId
+        }.toSet()
+        return reportIds
+    }
+
+    /**
+     * Submit a query to the History API, get a response, and do very basic sanity checks on it.
+     * @return Pair(pass/fail status, json body string)
+     * The json body may be null even if the test passed, in cases of an expected error code.
+     */
+    private fun historyApiQuery(testCase: SubmissionAPITestCase): Pair<Boolean, String?> {
+        val (request, response, result) = Fuel.get(testCase.path, testCase.parameters)
+            .authentication()
+            .bearer(testCase.bearer)
+            .responseString()
+        if (response.statusCode != testCase.expectedHttpStatus.value()) {
+            bad(
+                "***$name Test '${testCase.name}' FAILED:" +
+                    " Expected HttpStatus ${testCase.expectedHttpStatus}. Got ${response.statusCode}"
+            )
+            return Pair(false, null)
+        }
+        if (testCase.expectedHttpStatus != HttpStatus.OK) {
+            good("$name test '${testCase.name}' passed:  Got expected http status ${testCase.expectedHttpStatus}")
+            return Pair(true, null)
+        }
+        if (result !is Result.Success) {
+            bad("***$name Test '${testCase.name}' FAILED:  Result is $result")
+            return Pair(false, null)
+        }
+        val json: String = result.value
+        if (json.isNullOrEmpty()) {
+            bad("***$name Test '${testCase.name}' FAILED: empty body")
+            return Pair(false, null)
+        }
+        return Pair(true, json)
+    }
+
+    /**
+     * Compare the list of reportIds we got from the original submissions vs response from history API
+     * @return whether the test passed
+     */
+    private fun checkHistorySummary(
+        testCase: SubmissionAPITestCase,
+        jsonHistory: String
+    ): Boolean {
+        val jsonMapper = jacksonObjectMapper()
+        jsonMapper.registerModule(JavaTimeModule())
+        val submissionsHistories = jsonMapper.readValue(jsonHistory, Array<ExpectedSubmissionHistory>::class.java)
+        if (submissionsHistories.size != testCase.expectedReports.size) {
+            return bad(
+                "*** $name: TEST '${testCase.name}' FAILED: " +
+                    "${testCase.expectedReports.size} reports submitted" +
+                    " but got ${submissionsHistories.size} submission histories."
+            )
+        }
+
+        val missingReportIds = testCase.expectedReports.minus(submissionsHistories.mapNotNull { it.id }.toSet())
```

Comment:
This is a very minimal test -- there are lots of other values that come back from the lineage history that we could use to compare to submitted values.